### PR TITLE
Add parsing for extra sanitizers from PySecSan

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Signed-off-by: David Korczynski <david@adalogics.com>

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
I'm unsure about the `expected_state = ''` line -- any assistance here is appreciated!

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py [hunk 1]
```diff
@@ -3353,6 +3353,18 @@ def test_dns_resolution(self):
                                   expected_state, expected_stacktrace,
                                   expected_security_flag)
 
+  def test_pysecsan_command_os_system(self):
+    """Test PySecSan command injection bug in os.system"""
+    data = self._read_test_data('pysecsan_command_os_system.txt')
+    expected_type = 'PySecSan'
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:3359]: Can we make this a bit more detailed about what exact type it is?

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py [hunk 2]
```diff
@@ -3353,6 +3353,18 @@ def test_dns_resolution(self):
...
+    expected_address = ''
+    expected_state = ''
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:3361]: This should be the top three frames picked from the Python stacktrace. 

Our existing Python frame parsing regex is here: https://github.com/google/clusterfuzz/blob/2f89cd16a3cb6f2f5493c62315ac1973aac1f864/src/clusterfuzz/stacktraces/constants.py#L335

File: src/clusterfuzz/stacktraces/constants.py [hunk 3]
```diff
@@ -99,6 +99,7 @@
     r'===BUG DETECTED: Arbitrary file open===')
 EXTRA_SANITIZERS_ARBITRARY_DNS = re.compile(
     r'===BUG DETECTED: Arbitrary domain name resolution===')
+EXTRA_SANITIZERS_PYSECSAN_PYSECSAN = re.compile(r'===BUG DETECTED: PySecSan:')
```
Comment (oliverchang) [inline src/clusterfuzz/stacktraces/constants.py:102]: "PYSECSAN" is repeated in the variable name?

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py (hunk 1 above)
Comment (DavidKorczynski) [reply src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:3359]: I added it as a "PySecSan" type to allow for more experimentation in PySecSan (as in different types of bug detectors) without having to modify Clusterfuzz whenever trying one -- wdyt?

File: src/clusterfuzz/stacktraces/constants.py (hunk 3 above)
Comment (oliverchang) [reply src/clusterfuzz/stacktraces/constants.py:102]: Bumping this comment :) The variable name seems to have some unnecessary repetition?

**Review by oliverchang — APPROVED** [review]
LGTM with just one comment. thanks!

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py (hunk 1 above)
Comment (oliverchang) [reply src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:3359]: For the future we will want more detailed crash types, but if this is to experiment quickly this sounds fine.

**Review by oliverchang — APPROVED** [review]
thanks!

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py (hunk 1 above)
Comment (DavidKorczynski) [reply src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:3359]: Sounds good

**Pull request merged**